src/Symfony/Component/Routing/Tests/Fixtures/dumper/url_matcher1.php

```diff
@@ -36,36 +36,38 @@ public function match($pathinfo)
 
 
         // foo
-        if (0 === strpos($pathinfo, '/foo') && preg_match('#^/foo/(?P<bar>baz|symfony)$#s', $pathinfo, $matches)) {
+        if (0 === strpos($pathinfo, '/foo/') && preg_match('#^/foo/(?P<bar>baz|symfony)$#s', $pathinfo, $matches)) {
```

Comment:
I figured the most benefit would be gained from creating groups with common prefixes, with the extra `/` in there the common prefix base length is longer, which in this grouping is beneficial. I also thought that the two most common cases of routes were either fully static or dynamic but without optional parameters, especially as the first parameter. By including the `/` there we allow those to share the base prefix since static router group all the way to the `/`  naturally.